doc/conf.py

```diff
@@ -484,15 +484,21 @@ def js_tag_with_cache_busting(js):
         # the server, but will be used as part of the key for caching by browsers
         # so when we do a new meso release the switcher will update "promptly" on
         # the stable and devdocs.
-        "json_url": f"https://matplotlib.org/devdocs/_static/switcher.json?{SHA}",
+        "json_url": (
+            "https://output.circle-artifacts.com/output/job/"
+            f"{os.environ['CIRCLE_WORKFLOW_JOB_ID']}/artifacts/"
+            f"{os.environ['CIRCLE_NODE_INDEX']}"
+            "/doc/build/html/_static/switcher.json" if CIRCLECI else
+            f"https://matplotlib.org/devdocs/_static/switcher.json?{SHA}"
+        ),
         "version_match": (
             # The start version to show. This must be in switcher.json.
             # We either go to 'stable' or to 'devdocs'
             'stable' if matplotlib.__version_info__.releaselevel == 'final'
             else 'devdocs')
     },
     "navbar_end": ["theme-switcher", "version-switcher", "mpl_icon_links"],
-    "secondary_sidebar_items": "page-toc.html",
+    "navbar_persistent": ["search-button"],
```

Comment:
I believe removing `"secondary_sidebar_items": "page-toc.html",` reintroduced the "Show source" button in the right sidebar 
![grafik](https://github.com/matplotlib/matplotlib/assets/2836374/7d518593-9c81-4b4b-94bf-58983df1cc9e)

This was explicitly deactivated in  #23919. Though I have to admit, I don't quite understand the mechanisms here. It should alternatively be possible to remove that link via [configuration](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/source-buttons.html#view-source-link).

Reply:
Thanks for pointing out the config option, lets see if it works...

Reply:
Yep, seems to work.